src/hooks/menuItemsHooks/useProblemPaneMenuItems.tsx

```diff
@@ -33,24 +34,32 @@ export function useProblemPaneMenuItems(width: number, height: number) {
   const activeStorage = useAppSelector(activeResourceStorageSelector);
   const lastResponse = useValidationSelector(state => state.lastResponse);
   const selectedProblem = useValidationSelector(state => state.validationOverview.selectedProblem?.problem ?? null);
-  const selectedProblemMonacoData = useValidationSelector(state => {
-    const selectedFrom = state.validationOverview.selectedProblem?.selectedFrom ?? '';
+  const selectedProblemMonacoData = useAppSelector(state => {
+    const selectedFrom = state.validation.validationOverview.selectedProblem?.selectedFrom ?? '';
     if (!selectedFrom) return undefined;
-    const monacoData = {filePath: '', range: {}, resourceId: ''};
+    const monacoData: {filePath: ''; range: {}; resourceId: ''; storage: ResourceStorage} = {
+      filePath: '',
+      range: {},
+      resourceId: '',
+      storage: 'local',
+    };
 
     if (selectedFrom === 'file') {
-      return {...monacoData, ...problemFilePathAndRangeSelector(state)};
+      return {...monacoData, ...problemFilePathAndRangeSelector(state.validation)};
     }
 
     if (selectedFrom === 'resource') {
-      return {...monacoData, ...problemResourceIdAndRangeSelector(state)};
+      return {...monacoData, ...problemResourceAndRangeSelector(state)};
     }
   }, isEqual);
 
   const resourceSelection: ResourceSelection | undefined = useMemo(() => {
     if (!selectedProblemMonacoData?.resourceId) return undefined;
 
-    return {type: 'resource', resourceIdentifier: {id: selectedProblemMonacoData.resourceId, storage: activeStorage}};
+    return {
+      type: 'resource',
+      resourceIdentifier: {id: selectedProblemMonacoData.resourceId, storage: selectedProblemMonacoData.storage},
+    };
   }, [activeStorage, selectedProblemMonacoData]);
```

Comment:
Can we remove this unnecessary dependency?

Reply:
good catch - thank you